packages/components/CONTRIBUTING.md

```diff
@@ -4,12 +4,133 @@ Thank you for taking the time to contribute.
 
 The following is a set of guidelines for contributing to the `@wordpress/components` package to be considered in addition to the general ones described in our [Contributing Policy](/CONTRIBUTING.md).
 
-## Examples
+## Core principles
 
-Each component needs to include an example in its README.md file to demonstrate the usage of the component.
+Contributions to the `@wordpress/components` package should follow a set of core principles and technical requirements.
 
-These examples can be consumed automatically from other projects in order to visualize them in their documentation. To ensure these examples are extractable, compilable and renderable, they should be structured in the following way:
+This set of guidelines should apply especially to newly introduced components. It is, in fact, possible that some of the older component don't respect some of these guidelines for legacy/compat reasons.
 
--   It has to be included in a `jsx` code block.
--   It has to work out-of-the-box. No additional code should be needed to have working the example.
--   It has to define a React component called `My<ComponentName>` which renders the example (i.e.: `MyButton`). Examples for the Higher Order Components should define a `MyComponent<ComponentName>` component (i.e.: `MyComponentWithNotices`).
+### Compatibility
+
+The `@wordpress/components` package includes several components which are relied upon by many developers across different projects. It is, therefore, very important to avoid introducing breaking changes as much as possible.
+
+In these situations, an alternative approach is to "soft-deprecate" the given legacy API. This is achieved by removing traces of the API from the docs, while the code keeps supporing it in the background (with or without a warning)
```

Comment:
When and why do we add or not add a deprecation warning?

Reply:
That's a great question and one that I don't have an answer to. That's why it would be great to collaborate and come up with a more objective way to determine what do to in these scenarios.

Reply:
cc @gziolo in case you have highlights about the soft deprecation strategy, in particular about if and when we should output a deprecation warning

Reply:
We need to update first all places in Gutenberg that use API that is about to be deprecated. When that is ready, we can add warnings. If we do it earlier then Browser Console will be printing all those warnings. In addition to that, it would also fail e2e tests.

Reply:
Thank you for the explanation! I've amended the paragraph and added those "steps" in [`37c949d` (#33960)](https://github.com/WordPress/gutenberg/pull/33960/commits/37c949dfe3421109c6d8663e67dd2cca60e292dc)